src/mca/pdl/pdlopen/pdl_pdlopen_module.c

```diff
@@ -175,7 +175,26 @@ static int pdlopen_foreachfile(const char *search_path,
 
         dp = opendir(dirs[i]);
         if (NULL == dp) {
-            ret = PMIX_ERR_IN_ERRNO;
+            if (0 == i) {
```

Comment:
So I did some testing. I think this PR (and it's upstream version) are doing the right thing now.

Assume I have three different paths:
 * **A** `/u/jjhursey/local/current-ompi-dev/lib/openmpi` : Directory exists, and contains components
 * **B** `/u/jjhursey/.openmpi/components` : Directory does not exist
 * **C** `/u/jjhursey/` : Directory exists, but contains no components

Here are the behaviors that I observed:

### With this branch (before Ralph's most recent change):

 * `A:B` (Open MPI default). Everything works fine.
 * `B:A` Error message displayed by this PR
```
-------------------------------------------------------
No usable directories were found in the provided path
when searching for available plugins. This usually indicates
either that the OMPI installation was moved, or that the
OPAL_PREFIX environmental variable is set and pointing
to an incorrect or non-existent location.

Please correct the situation and try again.
-------------------------------------------------------
```
 * `C:A:B` `mpirun` does not work correctly
```
shell$ mpirun -np 4 -mca mca_base_component_path C:A:B hello_c
mpirun: Error: unknown option "-np"
```

### Comparing that with OMPI `master`:
 * `A:B` (Open MPI default). Everything works fine.
 * `B:A` `mpirun` does not work correctly
```
shell$ mpirun -np 4 -mca mca_base_component_path C:A:B hello_c
mpirun: Error: unknown option "-np"
```
 * `C:A:B` `mpirun` does not work correctly
```
shell$ mpirun -np 4 -mca mca_base_component_path C:A:B hello_c
mpirun: Error: unknown option "-np"
```

### After Ralph's most recent change to the upstream PR

 * `A:B` (Open MPI default). Everything works fine.
 * `B:A` Everything works fine.
 * `B` Error message displayed by this PR
```
-------------------------------------------------------
No usable directories were found in the provided path
when searching for available plugins. This usually indicates
either that the OMPI installation was moved, or that the
OPAL_PREFIX environmental variable is set and pointing
to an incorrect or non-existent location.

Please correct the situation and try again.
-------------------------------------------------------
```
 * `C:A:B` Everything works fine.
 * `C:B` `mpirun` does not work correctly
```
shell$ mpirun -np 4 -mca mca_base_component_path C:A:B hello_c
mpirun: Error: unknown option "-np"
```